build-logic/convention/src/main/kotlin/AndroidHiltConventionPlugin.kt

```diff
@@ -14,25 +14,13 @@
  *   limitations under the License.
  */
 
-import com.google.samples.apps.nowinandroid.libs
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.kotlin.dsl.dependencies
 
-class AndroidHiltConventionPlugin : Plugin<Project> {
-    override fun apply(target: Project) {
-        with(target) {
-            with(pluginManager) {
-                apply("com.google.devtools.ksp")
-                apply("dagger.hilt.android.plugin")
-            }
-
-            dependencies {
-                "implementation"(libs.findLibrary("hilt.android").get())
-                "ksp"(libs.findLibrary("hilt.compiler").get())
-            }
-
-        }
+class AndroidHiltConventionPlugin : Plugin<Project> by HiltConventionPlugin(
```

Comment:
Why not just `AndroidHiltConventionPlugin : HiltConventionPlugin`? And make `HiltConventionPlugin` an `abstract class`?

Reply:
It's just that I always try to favor composition over inheritence.
But I can switch the current code to abtract class if you prefer, let me know 👍